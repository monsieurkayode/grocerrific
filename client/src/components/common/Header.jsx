import React from 'react';
import { node, shape } from 'prop-types';
import { withRouter } from 'react-router-dom';

import logo from '../../assets/images/shoping_cart.png';

const Header = ({ children, history }) => (
  <header>
    <div className="header__left">
      <img id="logo" src={logo} alt="Brand" />
      <div
        onKeyPress={() => {}}
        role="link"
        tabIndex="0"
        id="brand"
        className="text__bold"
        onClick={() => history.push('/')}
      >
        Grocerrific<sup>&trade;</sup>
      </div>
    </div>
    { children }
  </header>
);

Header.propTypes = {
  children: node.isRequired,
  history: shape({}).isRequired
};

export default withRouter(Header);
