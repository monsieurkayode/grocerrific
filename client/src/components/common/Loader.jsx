import React from 'react';
import { number } from 'prop-types';
import PreloaderIcon from 'react-preloader-icon';
import Puff from 'react-preloader-icon/loaders/Puff';

/**
 *
 * @function Loader
 *
 * @param {object} props
 *
 * @return {JSX} JSX
 */
const Loader = ({ size }) => (
  <div className="loader">
    <PreloaderIcon
      loader={Puff}
      size={size}
      strokeWidth={12}
      strokeColor="#4d75a1"
      duration={1200}
    />
  </div>
);

Loader.propTypes = {
  size: number.isRequired
};

export default Loader;
