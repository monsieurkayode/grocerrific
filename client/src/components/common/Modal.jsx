import React from 'react';
import { node, string, func } from 'prop-types';

const Modal = ({ children, title, closeModal }) => (
  <div className="modal">
    <div className="modal__main">
      <div className="heading">
        <h2>{ title }
          <span
            onKeyPress={() => {}}
            role="button"
            tabIndex="0"
            onClick={() => closeModal()}
            className="close"
          >
            &times;
          </span>
        </h2>
      </div>
      { children }
    </div>
  </div>
);

Modal.propTypes = {
  children: node.isRequired,
  title: string.isRequired,
  closeModal: func.isRequired
};

export default Modal;
