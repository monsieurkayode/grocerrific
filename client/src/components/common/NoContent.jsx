import React from 'react';
import { string } from 'prop-types';

const NoContent = ({ content }) => (
  <div className="no-content">
    <i className="material-icons">info</i>
    <h3 className="no-content-info">
      {content}
    </h3>
  </div>
);

NoContent.propTypes = {
  content: string.isRequired
};

export default NoContent;
