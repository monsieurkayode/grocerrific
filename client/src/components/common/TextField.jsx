import React from 'react';
import {
  oneOfType,
  string,
  number,
  func
} from 'prop-types';

const TextField = ({
  name,
  label,
  placeholder,
  type,
  error,
  value,
  field,
  onChange
}) => (
  <div className={field}>
    <label htmlFor={name}>{label}&nbsp;<span>*</span></label>
    <input
      type={type}
      name={name}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      className={error.length > 0 && 'alert'}
    />
    { error.length > 0 && <div className="error">{error}</div> }
  </div>
);

TextField.defaultProps = {
  type: 'text',
  field: 'form-field',
  value: '',
  error: ''
};

TextField.propTypes = {
  name: string.isRequired,
  label: string.isRequired,
  placeholder: string.isRequired,
  type: string,
  error: string,
  value: oneOfType([string, number]),
  field: string,
  onChange: func.isRequired
};

export default TextField;
