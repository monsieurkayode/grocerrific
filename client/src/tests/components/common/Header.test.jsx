import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../../components/common/Header';

const setup = () => {
  const props = {
    children: <p>Header</p>,
    history: {
      push: jest.fn()
    }
  };

  const wrapper = shallow(<Header {...props} />);

  return { wrapper, props };
};

describe('Header Component', () => {
  const { wrapper, props } = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call navigate to root page when the brand is clicked', () => {
    const brandLink = wrapper.find('.header__left div');
    brandLink.simulate('click');
    expect(props.history.push).toHaveBeenCalledWith('/');
  });
});
