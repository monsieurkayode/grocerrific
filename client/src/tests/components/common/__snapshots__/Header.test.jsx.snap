// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<header>
  <div
    className="header__left"
  >
    <img
      alt="Brand"
      id="logo"
      src="test-file-mock"
    />
    <div
      className="text__bold"
      id="brand"
      onClick={[Function]}
      role="link"
      tabIndex="0"
    >
      Grocerrific
      <sup>
        â„¢
      </sup>
    </div>
  </div>
  <p>
    Header
  </p>
</header>
`;
